tensorflow_addons/image/filters.py

```diff
@@ -204,3 +204,91 @@ def median_filter2d(
+    sigma: FloatTensorLike = 1,
+    padding: str = "REFLECT",
+    constant_values: TensorLike = 0,
+    name: Optional[str] = None,
+) -> FloatTensorLike:
+    """Perform Gaussian Blur.
+
+    Args:
+      image: Either a 2-D `Tensor` of shape `[height, width]`,
+        a 3-D `Tensor` of shape `[height, width, channels]`,
+        or a 4-D `Tensor` of shape `[batch_size, height, width, channels]`.
+      filter_shape: An `integer` or `tuple`/`list` of 2 integers, specifying
+        the height and width of the 2-D median filter. Can be a single integer
+        to specify the same value for all spatial dimensions.
+      sigma: Standard deviation of Gaussian.
+      padding: A `string`, one of "REFLECT", "CONSTANT", or "SYMMETRIC".
+        The type of padding algorithm to use, which is compatible with
+        `mode` argument in `tf.pad`. For more details, please refer to
+        https://www.tensorflow.org/api_docs/python/tf/pad.
+      constant_values: A `scalar`, the pad value to use in "CONSTANT"
+        padding mode.
+      name: A name for this operation (optional).
+    Returns:
+      3-D or 4-D `Tensor` of the same dtype as input.
+    Raises:
+      ValueError: If `image` is not 2, 3 or 4-dimensional,
+        if `padding` is other than "REFLECT", "CONSTANT" or "SYMMETRIC",
+        or if `filter_shape` is invalid or sigma<=0.
+    """
+    with tf.name_scope(name or "gaussian_filter2d"):
+        if sigma <= 0:
+            raise ValueError("Sigma should not be zero")
+        if padding not in ["REFLECT", "CONSTANT", "SYMMETRIC"]:
+            raise ValueError("Padding should be REFLECT, CONSTANT, OR SYMMETRIC")
+
+        image = tf.cast(image, tf.float32)
+        image = img_utils.to_4D_image(image)
+        channels = tf.shape(image)[3]
+        filter_shape = keras_utils.normalize_tuple(filter_shape, 2, "filter_shape")
+
+        gaussian_filter_x = _get_gaussian_kernel(sigma, filter_shape[1])
+        gaussian_filter_x = tf.cast(gaussian_filter_x, tf.float32)
+        gaussian_filter_x = tf.reshape(gaussian_filter_x, [1, filter_shape[1]])
+
+        gaussian_filter_y = _get_gaussian_kernel(sigma, filter_shape[0])
+        gaussian_filter_y = tf.reshape(gaussian_filter_y, [filter_shape[0], 1])
+        gaussian_filter_y = tf.cast(gaussian_filter_y, tf.float32)
+
+        gaussian_filter_2d = _get_gaussian_kernel_2d(
+            gaussian_filter_y, gaussian_filter_x
+        )
+        gaussian_filter_2d = tf.repeat(gaussian_filter_2d, channels)
+        gaussian_filter_2d = tf.reshape(
+            gaussian_filter_2d, [filter_shape[0], filter_shape[1], channels, 1]
+        )
+
+        image = _pad(
+            image, filter_shape, mode=padding, constant_values=constant_values,
+        )
+
+        conv_ops = tf.nn.depthwise_conv2d(
+            input=image,
+            filter=gaussian_filter_2d,
+            strides=(1, 1, 1, 1),
+            padding="VALID",
+        )
```

Comment:
Seems that we forget to do `from_4D_image` to retain the original dimension.
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/filters.py#L122